src/web/app/src/components/Posts/CopyButton.tsx

```diff
@@ -1,52 +1,46 @@
-import { useState, CSSProperties, MouseEvent } from 'react';
-import { createStyles } from '@material-ui/core';
-import { makeStyles } from '@material-ui/core/styles';
-import Button from '@material-ui/core/IconButton';
-import Paper from '@material-ui/core/Paper';
-import FileCopyIcon from '@material-ui/icons/FileCopy';
-import { Transition } from 'react-transition-group';
-import clsx from 'clsx';
+// eslint-disable-next-line no-use-before-define
+import React, { useState, MouseEvent } from 'react';
+import { Tooltip, IconButton, createStyles, Zoom } from '@material-ui/core';
+import { makeStyles, withStyles } from '@material-ui/core/styles';
+import CopyIcon from '@material-ui/icons/FileCopyOutlined';
+import Check from '@material-ui/icons/Check';
 
 const useStyles = makeStyles(() =>
   createStyles({
-    copyButton: {
-      position: 'absolute',
-      right: 0,
-      padding: '1rem',
-      marginRight: '1.5rem',
-      transitionDuration: '0.2s',
-      transitionTimingFunction: 'ease',
-      cursor: 'pointer',
-      animation: 'fade-in-out 200ms both',
+    copy: {
+      fontSize: 'inherit',
     },
-    icon: {
-      fontSize: '2rem',
+    check: {
+      fill: '#3fb950',
+      fontSize: 'inherit',
     },
-    copyBadge: {
-      position: 'absolute',
-      top: '50%',
-      right: 0,
-      transitionProperty: 'transform, opacity',
-      transitionDuration: '0.2s',
-      transitionTimingFunction: 'ease',
-      borderRadius: '5px',
-      padding: '3px',
+    iconBtn: {
+      padding: '5px',
+      color: 'inherit',
+      fontSize: 'inherit',
     },
   })
 );
 
+const ButtonTooltip = withStyles({
+  tooltip: {
+    fontSize: 'inherit',
+    margin: 0,
+  },
+})(Tooltip);
+
 type CopyButtonProps = {
   onClick: (e: MouseEvent) => void;
+  beforeCopyMessage: string;
+  afterCopyMessage: string;
 };
 
-const transition: { [state: string]: CSSProperties } = {
-  entered: { transform: 'translate(-70%, -50%)', opacity: 1 },
-  entering: { transform: 'translate(0, -50%)', opacity: 0 },
-  exited: { transform: `translate(0, -50%)`, opacity: 0 },
-  exiting: { transform: `translate(0, -50%)`, opacity: 0 },
-};
-
-const CopyButton = ({ onClick }: CopyButtonProps) => {
+const CopyButton = ({
+  onClick,
+  beforeCopyMessage,
+  afterCopyMessage,
+  ...allOtherProps
+}: CopyButtonProps & React.AllHTMLAttributes<HTMLDivElement>) => {
```

Comment:
Do you want to import `AllHTMLAttributes` above so you don't have to use `React.*`?